tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/pull/contenttype/HttpRequestTask.java

```diff
@@ -0,0 +1,68 @@
+package com.dotcms.api.client.pull.contenttype;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.RecursiveTask;
+
+/**
+ * Represents a task that performs HTTP requests concurrently using the Fork/Join framework. It
+ * extends the RecursiveTask class and returns a list of ContentType objects.
+ */
+public class HttpRequestTask extends RecursiveTask<List<ContentType>> {
+
+    private final ContentTypeFetcher contentTypeFetcher;
+
+    private final transient List<ContentType> contentTypes;
+
+    private static final int THRESHOLD = 10;
+
+    public HttpRequestTask(final List<ContentType> contentTypes,
+            final ContentTypeFetcher contentTypeFetcher) {
+        this.contentTypes = contentTypes;
+        this.contentTypeFetcher = contentTypeFetcher;
+    }
+
+    @Override
+    protected List<ContentType> compute() {
+
+        if (contentTypes.size() <= THRESHOLD) {
```

Comment:
I think we could have a more effective "divide and conquer" type of strategy here.
As it is now if we have only 10 then we do only one request that's totally fine... 
But I'm thinking in larger numbers let's say 250 CT  we could divide the list in bunches of 4 
or do something like 

```
final int partitionSize = Math
				.max((contentTypes.size() / maxThreads), 10);
```
then partition the list of ct 
and then divide the partition between a fixed number of tasks

Reply:
only suggestion. its up to you